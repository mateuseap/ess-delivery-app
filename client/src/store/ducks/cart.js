import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  cartRequest: [],
  getCart: [],
  updateCart: ["rest_id", "rest_name", "item", "amountToChange"],
  makeOrder: ["callback"],
  cartSuccess: ["data"],
  cartError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.CART_ERROR]: error,
  [Types.CART_REQUEST]: request,
  [Types.CART_SUCCESS]: success,
});
