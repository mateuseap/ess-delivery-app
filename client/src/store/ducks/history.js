import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  historyRequest: [],
  getHistory: ["dateFilter"],
  postHistory: ["data"],
  historySuccess: ["data"],
  historyError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.HISTORY_ERROR]: error,
  [Types.HISTORY_REQUEST]: request,
  [Types.HISTORY_SUCCESS]: success,
});
