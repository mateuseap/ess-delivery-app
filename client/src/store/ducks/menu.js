import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  menuRequest: [],
  getMenu: ["id"],
  menuSuccess: ["data"],
  menuError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.MENU_ERROR]: error,
  [Types.MENU_REQUEST]: request,
  [Types.MENU_SUCCESS]: success,
});
