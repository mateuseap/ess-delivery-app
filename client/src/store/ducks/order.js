import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  orderRequest: [],
  getOrderDetails: ["id"],
  orderSuccess: ["data"],
  orderError: ["err"],
  cancelOrder: ["id", "callback"],
  cancelOrderWatch: [],
  orderStatusWatchWorker: ["id"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

const cancelOrderWatch = (state) => state.merge({});

export default createReducer(INITIAL_STATE, {
  [Types.ORDER_ERROR]: error,
  [Types.ORDER_REQUEST]: request,
  [Types.ORDER_SUCCESS]: success,
  [Types.CANCEL_ORDER_WATCH]: cancelOrderWatch,
});
