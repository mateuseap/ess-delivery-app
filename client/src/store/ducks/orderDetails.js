import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  orderDetailsRequest: [],
  getOrderDetails: ["id"],
  orderDetailsSuccess: ["data"],
  orderDetailsError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.ORDER_DETAILS_ERROR]: error,
  [Types.ORDER_DETAILS_REQUEST]: request,
  [Types.ORDER_DETAILS_SUCCESS]: success,
});
