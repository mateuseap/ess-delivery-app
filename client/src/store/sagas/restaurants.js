import { call, put } from "@redux-saga/core/effects";
import { Creators } from "../ducks/restaurants";
import api from "../../services/api";
import { toastr } from "react-redux-toastr";

export default function* getRestaurants() {
  try {
    yield put(Creators.restaurantsRequest());
    const response = yield call(api.get, "/restaurants", {});

    if (response.data) {
      yield put(Creators.restaurantsSuccess(response.data));
    }
  } catch (err) {
    yield put(Creators.restaurantsError({ err }));
    toastr.error("Erro ao buscar restaurantes");
  }
}
