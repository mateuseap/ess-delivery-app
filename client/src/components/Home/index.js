import React, { Component } from "react";
import {
  HomeStyle,
  TextStyle,
  CardTitle,
  DishName,
  DishDescription,
  CardGroup,
  CardBody,
  DishImg,
  Card,
  BtnStyle,
} from "./styles";

import ReactLoading from "react-loading";

import { Link } from "react-router-dom";
import { Button } from "react-bootstrap";

import FadeIn from "react-fade-in";

import { connect } from "react-redux";
import { Creators as RestaurantsCreator } from "../../store/ducks/restaurants";

class Home extends Component {
  getCard(element) {
    return (
      <Card key={element.restId}>
        <DishImg src={element.dishPhoto}></DishImg>

        <CardBody className="mb-5">
          <CardTitle>{element.restName}</CardTitle>
          <DishName>{element.dishName}</DishName>
          <DishDescription>{element.dishDescription}</DishDescription>
          {/* Quando apertar esse botão, o usuário deve ser redirecionado a tela de fazer pedidos com esse restaurante selecionado */}
          <Link to={`/menu/${element.restId}`}>
            <BtnStyle>
              <Button variant="success" className="m-2" name="BOTAO">
                PEÇA JÁ!
              </Button>
            </BtnStyle>
          </Link>
        </CardBody>
      </Card>
    );
  }

  componentDidMount() {
    this.props.getRestaurants();
  }

  render() {
    const { restaurants } = this.props;

    return (
      <HomeStyle>
        {restaurants.loading ? (
          <ReactLoading
            type={"spin"}
            style={{
              position: "absolute",
              width: "10vw",
              top: "40%",
              left: "0",
              right: "0",
              margin: "auto",
            }}
          />
        ) : (
          <div>
            <FadeIn transitionDuration={800}>
              <TextStyle>
                Saboreie as deliciosas comidas dos nossos restaurantes
                parceiros!
              </TextStyle>
            </FadeIn>
            <FadeIn transitionDuration={1000}>
              <CardGroup>
                {restaurants.data.map((restaurant) => {
                  const dish =
                    restaurant.menu.options[restaurant.menu.destaqueIndex];
                  return this.getCard({
                    restId: restaurant.id,
                    dishPhoto: dish.photo,
                    restName: restaurant.name,
                    dishName: dish.name,
                    dishDescription: dish.description,
                  });
                })}
              </CardGroup>
            </FadeIn>
          </div>
        )}
      </HomeStyle>
    );
  }
}

const mapStateToProps = ({ restaurants }) => ({ restaurants });

export default connect(mapStateToProps, { ...RestaurantsCreator })(Home);
