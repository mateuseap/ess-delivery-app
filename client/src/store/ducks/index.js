import { combineReducers } from "redux";

import { reducer as toastr } from "react-redux-toastr";

import user from "./user";
import restaurants from "./restaurants";
import history from "./history";
import cart from "./cart";
import menu from "./menu";
import order from "./order";

export default combineReducers({
  user,
  restaurants,
  history,
  cart,
  menu,
  order,
  toastr,
});
