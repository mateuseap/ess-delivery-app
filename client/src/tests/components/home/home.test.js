import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Home from "../../../components/Home";
import { BrowserRouter } from "react-router-dom";
import { INITIAL_STATE, Types } from "../../../store/ducks/restaurants";
import createSagaMiddleware from "redux-saga";
import { restaurantsLoadedState } from "../dataUtils";

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

const sagaMiddleware = createSagaMiddleware();

const mockStore = configureStore([sagaMiddleware]);

Enzyme.configure({ adapter: new Adapter() });

describe("Home Component Integration Testing", () => {
  it("Renders loading correctly", () => {
    const store = mockStore({ restaurants: INITIAL_STATE });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    );

    const expectedActions = [{ type: Types.GET_RESTAURANTS }];
    expect(store.getActions()).toEqual(expectedActions);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Renders data correctly", () => {
    const store = mockStore({
      restaurants: restaurantsLoadedState,
    });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    );
    const expectedActions = [{ type: Types.GET_RESTAURANTS }];
    expect(store.getActions()).toEqual(expectedActions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
