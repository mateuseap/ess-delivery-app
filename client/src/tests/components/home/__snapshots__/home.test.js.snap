// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Integration Testing Renders data correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Home)>
        <Home
          getRestaurants={[Function]}
          restaurants={
            Object {
              "data": Array [
                Object {
                  "addresses": Array [
                    "Avenida Caxangá, 634",
                  ],
                  "id": 2,
                  "menu": Object {
                    "destaqueIndex": 0,
                    "options": Array [
                      Object {
                        "description": "Hamburguer monstro de dar água na boca. ELIASLOL, O MELHOR COZINHEIRO DE LIMOEIRO!",
                        "item_id": 1,
                        "name": "Ogro Burguer com Batata Média",
                        "photo": "https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg",
                        "price": 25.5,
                      },
                      Object {
                        "description": "Coquinha geladinha HMMMMM!",
                        "item_id": 2,
                        "name": "Coca-cola lata",
                        "photo": "https://images-americanas.b2w.io/produtos/4575311740/imagens/coquinha-coca-cola-lata-220ml-desde-1886-sabor/4575311740_1_large.jpg",
                        "price": 5.5,
                      },
                    ],
                  },
                  "name": "Elias",
                  "photo": "https://media-cdn.tripadvisor.com/media/photo-s/05/85/e1/f5/sala-ristorante.jpg",
                  "rates": Array [
                    Object {
                      "feedback_text": "",
                      "stars": 4.5,
                      "user_id": "1",
                    },
                  ],
                },
                Object {
                  "addresses": Array [
                    "Avenida Bernardo Vieira de Melo, 980",
                  ],
                  "id": 3,
                  "menu": Object {
                    "destaqueIndex": 1,
                    "options": Array [
                      Object {
                        "description": "Melhor parme de piedade e arredores!",
                        "item_id": 1,
                        "name": "Frango a parmegiana",
                        "photo": "https://photos.bigoven.com/recipe/hero/fil-de-frango-parmegiana-bcd93f.jpg",
                        "price": 20,
                      },
                      Object {
                        "description": "Frango empanado a moda da casa!",
                        "item_id": 2,
                        "name": "Frango a milanesa",
                        "photo": "https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400",
                        "price": 22,
                      },
                      Object {
                        "description": "Guaraná 1L",
                        "item_id": 3,
                        "name": "Guaraná 1L",
                        "photo": "https://a-static.mlcdn.com.br/1500x1500/refrigerante-guarana-original-antarctica-1-litro/lojasemix/1803p/b4b32c23dd9c6e4d77fa9c37daf6732e.jpg",
                        "price": 10,
                      },
                    ],
                  },
                  "name": "Daylol",
                  "photo": "https://media-cdn.tripadvisor.com/media/photo-s/16/3d/cb/2f/quintal-da-praia.jpg",
                  "rates": Array [
                    Object {
                      "feedback_text": "",
                      "stars": 4.5,
                      "user_id": "1",
                    },
                  ],
                },
                Object {
                  "addresses": Array [
                    "Avenida Abdias de Carvalho, 142",
                  ],
                  "id": 4,
                  "menu": Object {
                    "destaqueIndex": 0,
                    "options": Array [
                      Object {
                        "description": "Idol Chicken para você que é fã dos coreanos :D",
                        "item_id": 1,
                        "name": "Frango KPOPPER",
                        "photo": "https://s2.glbimg.com/0X_hlC9TWAJUt6CsKQoSKcQ7hvM=/620x345/e.glbimg.com/og/ed/f/original/2019/06/10/kpop_chickin_4.jpg",
                        "price": 50,
                      },
                      Object {
                        "description": "Garrafa 500ml",
                        "item_id": 2,
                        "name": "Água",
                        "photo": "https://farmaciaindiana.vteximg.com.br/arquivos/ids/206727/7896371000045_0.jpg?v=636748773745400000",
                        "price": 4,
                      },
                    ],
                  },
                  "name": "Alicelol",
                  "photo": "https://vejasp.abril.com.br/wp-content/uploads/2016/12/k-pop-chiken2.jpeg",
                  "rates": Array [
                    Object {
                      "feedback_text": "",
                      "stars": 4.5,
                      "user_id": "1",
                    },
                  ],
                },
              ],
              "error": false,
              "loading": false,
            }
          }
          restaurantsError={[Function]}
          restaurantsRequest={[Function]}
          restaurantsSuccess={[Function]}
        >
          <styled.div>
            <div
              className="sc-bczRLJ UWzdy"
            >
              <div>
                <FadeIn
                  transitionDuration={800}
                >
                  <div>
                    <div
                      key=".0"
                      style={
                        Object {
                          "opacity": 0,
                          "transform": "translateY(20px)",
                          "transition": "opacity 800ms, transform 800ms",
                        }
                      }
                    >
                      <styled.h1>
                        <h1
                          className="sc-gsnTZi gouaEI"
                        >
                          Saboreie as deliciosas comidas dos nossos restaurantes parceiros!
                        </h1>
                      </styled.h1>
                    </div>
                  </div>
                </FadeIn>
                <FadeIn
                  transitionDuration={1000}
                >
                  <div>
                    <div
                      key=".0"
                      style={
                        Object {
                          "opacity": 0,
                          "transform": "translateY(20px)",
                          "transition": "opacity 1000ms, transform 1000ms",
                        }
                      }
                    >
                      <styled.div>
                        <div
                          className="sc-dkzDqf hAmERx"
                        >
                          <styled.div
                            key="2"
                          >
                            <div
                              className="sc-hKMtZM jEdtfJ"
                            >
                              <styled.div
                                src="https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg"
                              >
                                <div
                                  className="sc-ftvSup bXYdD"
                                  src="https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg"
                                />
                              </styled.div>
                              <styled.div
                                className="mb-5"
                              >
                                <div
                                  className="sc-iBkjds UQhGq mb-5"
                                >
                                  <styled.div>
                                    <div
                                      className="sc-eCYdqJ itztyL"
                                    >
                                      Elias
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-jSMfEi jEMxCd"
                                    >
                                      Ogro Burguer com Batata Média
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-gKXOVf gZPvUT"
                                    >
                                      Hamburguer monstro de dar água na boca. ELIASLOL, O MELHOR COZINHEIRO DE LIMOEIRO!
                                    </div>
                                  </styled.div>
                                  <Link
                                    to="/menu/2"
                                  >
                                    <a
                                      href="/menu/2"
                                      onClick={[Function]}
                                    >
                                      <styled.div>
                                        <div
                                          className="sc-papXJ iESUCh"
                                        >
                                          <Button
                                            active={false}
                                            className="m-2"
                                            disabled={false}
                                            name="BOTAO"
                                            variant="success"
                                          >
                                            <button
                                              className="m-2 btn btn-success"
                                              disabled={false}
                                              name="BOTAO"
                                              type="button"
                                            >
                                              PEÇA JÁ!
                                            </button>
                                          </Button>
                                        </div>
                                      </styled.div>
                                    </a>
                                  </Link>
                                </div>
                              </styled.div>
                            </div>
                          </styled.div>
                          <styled.div
                            key="3"
                          >
                            <div
                              className="sc-hKMtZM jEdtfJ"
                            >
                              <styled.div
                                src="https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400"
                              >
                                <div
                                  className="sc-ftvSup bEpkvJ"
                                  src="https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400"
                                />
                              </styled.div>
                              <styled.div
                                className="mb-5"
                              >
                                <div
                                  className="sc-iBkjds UQhGq mb-5"
                                >
                                  <styled.div>
                                    <div
                                      className="sc-eCYdqJ itztyL"
                                    >
                                      Daylol
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-jSMfEi jEMxCd"
                                    >
                                      Frango a milanesa
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-gKXOVf gZPvUT"
                                    >
                                      Frango empanado a moda da casa!
                                    </div>
                                  </styled.div>
                                  <Link
                                    to="/menu/3"
                                  >
                                    <a
                                      href="/menu/3"
                                      onClick={[Function]}
                                    >
                                      <styled.div>
                                        <div
                                          className="sc-papXJ iESUCh"
                                        >
                                          <Button
                                            active={false}
                                            className="m-2"
                                            disabled={false}
                                            name="BOTAO"
                                            variant="success"
                                          >
                                            <button
                                              className="m-2 btn btn-success"
                                              disabled={false}
                                              name="BOTAO"
                                              type="button"
                                            >
                                              PEÇA JÁ!
                                            </button>
                                          </Button>
                                        </div>
                                      </styled.div>
                                    </a>
                                  </Link>
                                </div>
                              </styled.div>
                            </div>
                          </styled.div>
                          <styled.div
                            key="4"
                          >
                            <div
                              className="sc-hKMtZM jEdtfJ"
                            >
                              <styled.div
                                src="https://s2.glbimg.com/0X_hlC9TWAJUt6CsKQoSKcQ7hvM=/620x345/e.glbimg.com/og/ed/f/original/2019/06/10/kpop_chickin_4.jpg"
                              >
                                <div
                                  className="sc-ftvSup uZhNT"
                                  src="https://s2.glbimg.com/0X_hlC9TWAJUt6CsKQoSKcQ7hvM=/620x345/e.glbimg.com/og/ed/f/original/2019/06/10/kpop_chickin_4.jpg"
                                />
                              </styled.div>
                              <styled.div
                                className="mb-5"
                              >
                                <div
                                  className="sc-iBkjds UQhGq mb-5"
                                >
                                  <styled.div>
                                    <div
                                      className="sc-eCYdqJ itztyL"
                                    >
                                      Alicelol
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-jSMfEi jEMxCd"
                                    >
                                      Frango KPOPPER
                                    </div>
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="sc-gKXOVf gZPvUT"
                                    >
                                      Idol Chicken para você que é fã dos coreanos :D
                                    </div>
                                  </styled.div>
                                  <Link
                                    to="/menu/4"
                                  >
                                    <a
                                      href="/menu/4"
                                      onClick={[Function]}
                                    >
                                      <styled.div>
                                        <div
                                          className="sc-papXJ iESUCh"
                                        >
                                          <Button
                                            active={false}
                                            className="m-2"
                                            disabled={false}
                                            name="BOTAO"
                                            variant="success"
                                          >
                                            <button
                                              className="m-2 btn btn-success"
                                              disabled={false}
                                              name="BOTAO"
                                              type="button"
                                            >
                                              PEÇA JÁ!
                                            </button>
                                          </Button>
                                        </div>
                                      </styled.div>
                                    </a>
                                  </Link>
                                </div>
                              </styled.div>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                    </div>
                  </div>
                </FadeIn>
              </div>
            </div>
          </styled.div>
        </Home>
      </Connect(Home)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Home Component Integration Testing Renders loading correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Home)>
        <Home
          getRestaurants={[Function]}
          restaurants={
            Object {
              "data": Array [],
              "error": false,
              "loading": false,
            }
          }
          restaurantsError={[Function]}
          restaurantsRequest={[Function]}
          restaurantsSuccess={[Function]}
        >
          <styled.div>
            <div
              className="sc-bczRLJ UWzdy"
            >
              <div>
                <FadeIn
                  transitionDuration={800}
                >
                  <div>
                    <div
                      key=".0"
                      style={
                        Object {
                          "opacity": 0,
                          "transform": "translateY(20px)",
                          "transition": "opacity 800ms, transform 800ms",
                        }
                      }
                    >
                      <styled.h1>
                        <h1
                          className="sc-gsnTZi gouaEI"
                        >
                          Saboreie as deliciosas comidas dos nossos restaurantes parceiros!
                        </h1>
                      </styled.h1>
                    </div>
                  </div>
                </FadeIn>
                <FadeIn
                  transitionDuration={1000}
                >
                  <div>
                    <div
                      key=".0"
                      style={
                        Object {
                          "opacity": 0,
                          "transform": "translateY(20px)",
                          "transition": "opacity 1000ms, transform 1000ms",
                        }
                      }
                    >
                      <styled.div>
                        <div
                          className="sc-dkzDqf hAmERx"
                        />
                      </styled.div>
                    </div>
                  </div>
                </FadeIn>
              </div>
            </div>
          </styled.div>
        </Home>
      </Connect(Home)>
    </Router>
  </BrowserRouter>
</Provider>
`;
